plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notifications
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.RoamingType
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.util.Alarm
+import com.intellij.util.AlarmFactory
+import com.intellij.util.io.HttpRequests
+import software.aws.toolkits.core.utils.RemoteResolveParser
+import software.aws.toolkits.core.utils.RemoteResource
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.core.DefaultRemoteResourceResolverProvider
+import software.aws.toolkits.jetbrains.core.RemoteResourceResolverProvider
+import software.aws.toolkits.telemetry.Component
+import software.aws.toolkits.telemetry.ToolkitTelemetry
+import java.io.InputStream
+import java.time.Duration
+import java.util.concurrent.atomic.AtomicBoolean
+
+private const val NOTIFICATION_ENDPOINT = "" // TODO: Replace with actual endpoint
+private const val MAX_RETRIES = 3
+private const val RETRY_DELAY_MS = 1000L
+
+object NotificationFileValidator : RemoteResolveParser {
+    override fun canBeParsed(data: InputStream): Boolean =
+        try {
+            NotificationMapperUtil.mapper.readValue<NotificationsList>(data)
+            true
+        } catch (e: Exception) {
+            false
+        }
+}
+
+@State(name = "NotificationETagState", storages = [Storage("aws.xml", roamingType = RoamingType.DISABLED)])
+object ETagState : PersistentStateComponent<ETagState.State> {
+    data class State(
```

Comment:
This return type could lie outside this class.
nit: maybe add a more descriptive name.
eg reference class: EcsExecCommandSettings